manipmongo/internal/compiler/filter_test.go

```diff
@@ -14,12 +14,39 @@ package compiler
 import (
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/globalsign/mgo/bson"
 	. "github.com/smartystreets/goconvey/convey"
 	"go.aporeto.io/elemental"
 )
 
+func toMap(in bson.D) bson.M {
+
+	out := make(bson.M, len(in))
+
+	for _, item := range in {
+
+		switch iv := item.Value.(type) {
+
+		case bson.D:
+			out[item.Name] = toMap(iv)
+
+		case []bson.D:
+			outs := make([]bson.M, len(iv))
+			for i, subitem := range iv {
+				outs[i] = toMap(subitem)
+			}
+			out[item.Name] = outs
+
+		default:
+			out[item.Name] = item.Value
+		}
+	}
+
+	return out
+}
```

Comment:
The `bson.D` type has a `Map()` method, however I am guessing you added this helper because it doesn't traverse the structure to convert any nested `bson.D`'s to `bson.D` which would make the JSON output ugly?

Reply:
Alternatively, for unit testing purposes, you could just do a `reflect.DeepEqual` to the expected `bson.M` type instead of serializing to JSON.

Reply:
Yes. It was not recursive. And I did not want to change the tests to make sure I did not change anything